[WFLY-13771]: Adding messaging layers and some testing

ehsavoie opened the pull request:
https://issues.redhat.com/browse/WFLY-13771

This is using the commits from https://github.com/wildfly/wildfly/pull/13524

bstansberry committed:
WFLY-13354 Add Galleon layers for the EJB subsystem

bstansberry commented:
Superseded by #13535 which is this commit on top of the current EJB layers branch.

bstansberry reviewed (changes requested):
@ehsavoie This needs an update to update https://github.com/wildfly/wildfly/pull/13527/files#diff-9a6f6b7a6bd30fddb786db3840aac00f

tommaso-borgato commented on testsuite/layers/pom.xml:
```diff
@@ -2161,6 +2381,8 @@
                                                 <layer>ee</layer>
                                                 <layer>ee-security</layer>
+                                                <layer>ejb</layer>
+                                                <layer>ejb-lite</layer>
```
@ehugonne I can't see the `remote-activemq` added to `all-layers-provisioning-full` as written in the TP (https://gitlab.mw.lab.eng.bos.redhat.com/jbossqe-eap/test-plans/merge_requests/392/diffs#1135f3443166464178d2031722561a56e1bdc307_0_105)

tommaso-borgato commented on testsuite/layers/pom.xml:
```diff
@@ -2250,6 +2472,8 @@
                                                 <layer>ee</layer>
                                                 <layer>ee-security</layer>
+                                                <layer>ejb</layer>
+                                                <layer>ejb-lite</layer>
```
@ehugonne I can't see the `remote-activemq` added to `all-layers-jpa-distributed-provisioning-full` as written in the TP (https://gitlab.mw.lab.eng.bos.redhat.com/jbossqe-eap/test-plans/merge_requests/392/diffs#1135f3443166464178d2031722561a56e1bdc307_0_106)

tommaso-borgato commented:
@ehsavoie I see may modifications to existing JMS tests; I didn't find modifications to existing tests in the other EAP7-1444 related RFEs and I couldn't find motivations for these modifications in AD and TP: can you to provide some details on this? is it some refactoring/improvements or something necessary to make tests work also when a slimmed version of the server is used (I mean wildfly with the `remote-activemq` + some other layers)?

ehsavoie commented:
@tommaso-borgato the changes are due to enable the same code to use an embedded broker and a remote one. For example with a remote broker you can't rely on a WildFly management operation to get the message count in a topic.

ehsavoie committed:
[WFLY-13771]: Adding messaging layers and some testing

  * Simplifying remote broker testing
  * Defining 3 layers for messaging :
	- messaging-activemq
	- remote-activemq
	- embedded-activemq (for tests only).
  * Adding support to start / stop remote broker

Jira: https://issues.redhat.com/browse/WFLY-13771

ehsavoie commented:
The failure on windows don't seem related.

bstansberry commented:
@ehsavoie Please rebase

bstansberry commented:
@ehsavoie if you're around though, please wait a bit as I'll merge others that may produce more conflicts

bstansberry commented:
@ehsavoie I see this has been through pre-check, so rather than you rebasing I'll do it on top of the other layers branches and I'll push that up and merge the whole set.  @jimma can then base on my branch while you sleep and not wait for you to rebase tomorrow.

bstansberry commented:
Superseded by #13546
